lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -1701,9 +1708,19 @@ class BuildControl : public FuncOpPartialLoweringPattern {
                  whileSchedPtr) {
         auto &whileOp = whileSchedPtr->whileOp;
 
-        /// Insert while iter arg initialization group.
-        rewriter.create<calyx::EnableOp>(loc,
-                                         whileSchedPtr->initGroup.getName());
+        /// Insert while iter arg initialization group. For a single iter arg
+        /// initialization, just emit an enable, but for multiple, emit a
+        /// parallel group to assign them all at once.
```

Comment:
Do we need this kind of special case? I could imagine that a `par` group containing only a single `calyx.enable` nested within a `seq` should be canonicalized such that the `calyx.enable` is lifted to the outer `seq`?

Reply:
Fair enough, I think always using a par group makes sense.